### ISSUE TITLE
throw error for reserved words

### ISSUE THREAD

**Issue by zachvalenta** [alanmcruickshank/sqlfluff#194]
if I incorrectly use a reserved word in a DDL statement:
```sql
-- SQLite
CREATE TABLE artist(artist_name TEXT);  -- ok
CREATE TABLE artist(create TEXT);  -- `create` is a reserved word, syntax error
```

...should sqlfluff throw a lint error? Right now now it doesn't.

**Comment by alanmcruickshank** [comment]
Thanks @zachvalenta - some reserved words will cause a parsing error at the moment so will get caught, but I suspect that there is a subset of reserved words that don't. Do you have access to a canonical list of reserved words for SQLite?

**Comment by zachvalenta** [comment]
> Do you have access to a canonical list of reserved words for SQLite?

Here's an [ANSI list](https://www.sqlstyle.guide/#reserved-keyword-reference). 

Most dbms seem to have their own twist on what certain keywords (reserved and non-reserved) mean within their system.

> In the PostgreSQL parser life is a bit more complicated. There are several different classes of tokens ranging from those that can never be used as an identifier to those that have absolutely no special status in the parser...even reserved key words are not completely reserved in PostgreSQL - https://www.postgresql.org/docs/8.1/sql-keywords-appendix.html

So this is admittedly not a straightforward thing for a linter to deal with :)

`create` though should be reserved across SQLite, Postgres, et al.

**Comment by Katzmann1983** [comment]
@alanmcruickshank  First thing, which comes to mind is to add them in the NakedIdentifierSegment _anti_template
I can do that, bby adding a list of these words and dynamically creating the anti_template.
Additionally, the keywords could all be KeywordSegments which are also generated somehow on the fly. Let me know, what you think, and I will give it a try :)
